### ISSUE TITLE
java.lang.Exception: Clj-kondo cache is locked by other thread or process.

### ISSUE THREAD

**Issue by stig** [clj-kondo/clj-kondo#2408]
[ To keep development of this project going, consider sponsoring. If you are
already a sponsor, thank you! ]

- [ ] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**version**

clj-kondo v2024.03.13

**macro usage**

N/A

[ Is your bug related to macro usage? Consider using a configuration. If not, ignore this section. ]

See [config.md](https://github.com/clj-kondo/clj-kondo/blob/master/doc/config.md#unrecognized-macros) for related configuration options.

**platform**

MacOS. I'm using a version of clj-kondo installed via nixpkgs.

**editor**

I ran this on the commandline, but I ran it in Emacs' Eshell, and I have clojure-lsp running too.

**problem**

I ran `clj-kondo --lint src:test` and it spat out this error:

```
Unexpected error. Please report an issue.
java.lang.Exception: Clj-kondo cache is locked by other thread or process.
	at clj_kondo.impl.cache$sync_cache.invokeStatic(cache.clj:192)
	at clj_kondo.core$run_BANG_.invokeStatic(core.clj:214)
	at clj_kondo.main$main.invokeStatic(main.clj:139)
	at clj_kondo.main$main.doInvoke(main.clj:135)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clj_kondo.main$_main$fn__14015.invoke(main.clj:186)
	at clj_kondo.main$_main.invokeStatic(main.clj:186)
	at clj_kondo.main$_main.doInvoke(main.clj:184)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clj_kondo.main.main(Unknown Source)
	at java.base@22/java.lang.invoke.LambdaForm$DMH/sa346b79c.invokeStaticInit(LambdaForm$DMH)
```

**repro**

[ Please provide a minimal and complete reproduction of the problem, including a namespace form, which can be pasted in a repro.clj file. Please include the output of the invocation of `clj-kondo --lint repro.clj` (or `.cljs/.cljc`) from the command line. ]

I can't reproduce it. It happened once running on a 130,000 line project.

**config**

I don't think the configuration is relevant. I reran the same command and it ran successfully.

**expected behavior**

[ What is the behavior you expected to see from clj-kondo? ]

No crash!

[ Optional: if you are interested in doing a PR yourself, please leave a note. ]

No idea where to start, sorry. It's the first time I've had a crash from clj-kondo and may not have even bothered to report it if it didn't ask me to.

**Comment by borkdude** [comment]
I did fix something related recently so this issue might be fixed. If you may have a repro in the future, please tell me and we'll re-open this issue.
